lib/vrl/stdlib/src/replace.rs

```diff
@@ -89,14 +89,20 @@ struct ReplaceFn {
 impl Expression for ReplaceFn {
     fn resolve(&self, ctx: &mut Context) -> Resolved {
         let value = self.value.resolve(ctx)?;
+        let value = value.borrow();
         let value = value.try_bytes_utf8_lossy()?;
 
         let with_value = self.with.resolve(ctx)?;
+        let with_value = with_value.borrow();
```

Comment:
This is a common pattern. Unfortunately it seems we can't add this function to `SharedValue` like:

```rust
pub fn try_bytes_utf8_lossy(&self) -> Result<Cow<'_, str>, crate::value::error::Error> {
        let value = self.borrow();
        value.try_bytes_utf8_lossy()
}
```

Since that results in an error: 

```
error[E0515]: cannot return value referencing local variable `value`
  --> lib/vrl/compiler/src/shared_value.rs:72:9
   |
72 |         value.try_bytes_utf8_lossy()
   |         -----^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         returns a value referencing data owned by the current function
   |         `value` is borrowed here
```


(It does work for Interer however, since that is Copy. I will add that.)